lib/default.nix

```diff
@@ -393,4 +403,115 @@ in {
+              (_: { doCoverage = lib.mkDefault true; });
+          }];
+        }
+      ];
+    in
+      builtins.listToAttrs (lib.concatMap (packageName: [{
+        name = packageName;
+        value = coverageProject.hsPkgs.${packageName}.coverageReport;
+      }]) (packageNames coverageProject));
+
+  mkFlakePackages = haskellPackages: builtins.listToAttrs (
+    lib.concatLists (lib.mapAttrsToList (packageName: package:
+        lib.optional (package.components ? library)
+            { name = "${packageName}:lib:${packageName}"; value = package.components.library; }
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:lib:${n}"; value = v; })
+          (package.components.sublibs)
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:exe:${n}"; value = v; })
+          (package.components.exes)
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:test:${n}"; value = v; })
+          (package.components.tests)
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:bench:${n}"; value = v; })
+          (package.components.benchmarks)
+    ) haskellPackages));
+
+  mkFlakeApps = haskellPackages: builtins.listToAttrs (
+    lib.concatLists (lib.mapAttrsToList (packageName: package:
+      lib.mapAttrsToList (n: v:
+            { name = "${packageName}:exe:${n}"; value = { type = "app"; program = v.exePath; }; })
+          (package.components.exes)
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:test:${n}"; value = { type = "app"; program = v.exePath; }; })
+          (package.components.tests)
+        ++ lib.mapAttrsToList (n: v:
+            { name = "${packageName}:benchmark:${n}"; value = { type = "app"; program = v.exePath; }; })
+          (package.components.benchmarks)
+    ) haskellPackages));
+
+  mkFlakeCiJobs = project: {
+          checks
+        , coverage
+        , devShells
+        , checkedProject
+    }: {
+      # Run all the tests and code coverage
+      # Also build and cache any tools in the `devShells`
+      inherit checks coverage devShells;
+      # Build tools and cache tools needed for the project
+      inherit (project) roots;
+    }
+    # Build the plan-nix and check it if materialized
+    // lib.optionalAttrs (checkedProject ? plan-nix) {
+      plan-nix = checkedProject.plan-nix;
+    }
+    # Build the stack-nix and check it if materialized
+    // lib.optionalAttrs (checkedProject ? stack-nix) {
+      stack-nix = checkedProject.stack-nix;
+    };
+
+  mkFlake = project: {
+          selectPackages ? haskellLib.selectProjectPackages
+        , haskellPackages ? selectPackages project.hsPkgs
+        , packages ? mkFlakePackages haskellPackages
+        , apps ? mkFlakeApps haskellPackages
+        , coverageProjectModule ? {}
+        , checks ? flattenChecks (collectChecks' haskellPackages)
+        , coverage ? projectCoverageCiJobs project selectPackages coverageProjectModule
```

Comment:
I guess my only question is what we want the default to be here. Building with coverage is quite expensive (I think?) since you basically rebuild the entire project. So our users might get a bit annoyed that the default flake makes them do all this work if they don't care about coverage.

Reply:
I've switched it off by default now